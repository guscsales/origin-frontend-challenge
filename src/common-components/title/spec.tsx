import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import { create } from 'react-test-renderer';
import theme from '../../configs/styles/themes';
import { ThemeProvider } from 'styled-components';
import Title from '.';

describe('Title', () => {
  const getWrapper = (props: any = {}, renderType: any = mount) =>
    renderType(
      <ThemeProvider theme={theme}>
        <Title level={1} {...props}>
          test
        </Title>
      </ThemeProvider>
    );

  describe('#render', () => {
    it('should match a snapshot', () => {
      const wrapper = getWrapper({}, create).toJSON();
      expect(wrapper).toMatchSnapshot();
    });

    it('should render a correct heading', () => {
      const wrapper: ReactWrapper = getWrapper();

      expect(wrapper.find('h1')).toHaveLength(1);
    });
  });
});
