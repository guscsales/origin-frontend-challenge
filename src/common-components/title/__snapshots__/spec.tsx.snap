// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title #render should match a snapshot 1`] = `
.c0 {
  font-family: Work Sans;
  font-size: 2.8rem;
  font-weight: 600;
  margin: 0;
  margin-bottom: 4rem;
  color: #1b31a8;
}

.c0 strong {
  font-weight: 600;
}

.c0 span {
  display: block;
  font-size: 1.5rem;
  font-weight: 400;
  margin-top: 0.4rem;
  color: #657786;
}

@media (min-width:1024px) {
  .c0 {
    margin: 0 0 4.5rem;
  }
}

<h1
  className="c0"
>
  test
</h1>
`;
